api/controlplane/kubeadm/v1beta2/kubeadm_control_plane_types.go

```diff
@@ -758,7 +758,7 @@ type KubeadmControlPlaneV1Beta1DeprecatedStatus struct {
 	// Deprecated: This field is deprecated and is going to be removed when support for v1beta1 will be dropped. Please see https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20240916-improve-status-in-CAPI-resources.md for more details.
 	//
 	// +optional
-	UnavailableReplicas int32 `json:"unavailableReplicas"`
+	UnavailableReplicas int32 `json:"unavailableReplicas"` //nolint:kubeapilinter // field will be removed when v1beta1 is removed
```

Comment:
Just a note, when we are doing `noline:kubeapilinter` in the future, it's better to use the regex based excludes so that we can exclude single messages rather than entire fields from being checked across all rules.

In this case since it's going away, this is fine though

Reply:
I thought there is less harm in a local nolint vs an exclude which might match other fields accidentally

Reply:
I _think_ you can scope at least to package level and field name, so provided there aren't fields with duplicated names (which there probably are for us) then that would work. Would have to double check if we can scope more succinctly